cmd/cost/get_test.go

```diff
@@ -0,0 +1,117 @@
+package cost
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/organizations/types"
+	"github.com/onsi/gomega"
+	"github.com/shopspring/decimal"
+	"github.com/stretchr/testify/require"
+)
+
+var mockNowFunc = time.Now
+
+func TestGetTimePeriod(t *testing.T) {
+	refTime := time.Now()
+	mockNowFunc = func() time.Time { return refTime }
+
+	tests := []struct {
+		name          string
+		timePtr       string
+		expectedStart string
+		expectedEnd   string
+	}{
+		{
+			name:          "Default",
+			timePtr:       "",
+			expectedStart: fmt.Sprintf("%d-%02d-%02d", refTime.Year()-1, refTime.Month(), 01),
+			expectedEnd:   fmt.Sprintf("%d-%02d-%02d", refTime.Year(), refTime.Month(), refTime.Day()),
+		},
+		{
+			name:          "LM",
+			timePtr:       "LM",
+			expectedStart: fmt.Sprintf("%d-%02d-%02d", refTime.Year(), refTime.Month()-1, 01),
+			expectedEnd:   fmt.Sprintf("%d-%02d-%02d", refTime.Year(), refTime.Month(), 01),
+		},
+		{
+			name:          "MTD",
+			timePtr:       "MTD",
+			expectedStart: fmt.Sprintf("%d-%02d-%02d", refTime.Year(), refTime.Month(), 01),
+			expectedEnd:   fmt.Sprintf("%d-%02d-%02d", refTime.Year(), refTime.Month(), refTime.Day()),
+		},
+		{
+			name:          "YTD",
+			timePtr:       "YTD",
+			expectedStart: fmt.Sprintf("%d-%02d-%02d", refTime.Year(), 01, 01),
+			expectedEnd:   fmt.Sprintf("%d-%02d-%02d", refTime.Year(), refTime.Month(), refTime.Day()),
+		},
+		{
+			name:          "1Y",
+			timePtr:       "1Y",
+			expectedStart: refTime.AddDate(-1, 0, 0).Format("2006-01-02"),
+			expectedEnd:   fmt.Sprintf("%d-%02d-%02d", refTime.Year(), refTime.Month(), refTime.Day()),
+		},
+	}
```

Comment:
We should also be testing for expected failures, such as:
-  `asdfadsfdd`
- `mtd` (test for case insensitivity.  should the method being tested automatically uppercase the flag?) 
- "3Y"
- etc